src/Impostor.Server/Events/Game/Player/PlayerMovementEvent.cs

```diff
@@ -1,24 +1,71 @@
-﻿using Impostor.Api.Events.Player;
+using System;
+using System.Collections.Concurrent;
+
+using Impostor.Api.Events.Player;
 using Impostor.Api.Games;
 using Impostor.Api.Net;
 using Impostor.Api.Net.Inner.Objects;
+using Impostor.Hazel;
 
 namespace Impostor.Server.Events.Player
 {
     // TODO: Finish and use event, needs to be pooled
-    public class PlayerMovementEvent : IPlayerEvent
+    public class PlayerMovementEvent : IPlayerMovementEvent , IRecyclable , IDisposable
     {
+        private static readonly ConcurrentBag<PlayerMovementEvent> items = new ConcurrentBag<PlayerMovementEvent>();
+        private static int counter = 0;
+        private static int MAX = 25;
+
         public PlayerMovementEvent(IGame game, IClientPlayer clientPlayer, IInnerPlayerControl playerControl)
         {
             Game = game;
             ClientPlayer = clientPlayer;
             PlayerControl = playerControl;
         }
 
-        public IGame Game { get; }
+        public IGame Game { get; private set; }
+
+        public IClientPlayer ClientPlayer { get; private set; }
 
-        public IClientPlayer ClientPlayer { get; }
+        public IInnerPlayerControl PlayerControl { get; private set; }
 
-        public IInnerPlayerControl PlayerControl { get; }
+        public static PlayerMovementEvent Get(IGame game, IClientPlayer clientPlayer, IInnerPlayerControl playerControl)
```

Comment:
Could there be a better method name for this? Also, why is this a static method?

Reply:
This method checks if there is an existing instance of PLayerMovmentEvent to use it instead of constructing a new object.
It is static to be easier to use anywhere.